terratorch/tasks/regression_tasks.py

```diff
@@ -115,6 +117,285 @@ def plot(self, val: Tensor | Sequence[Tensor] | None = None, ax=None):
+            return {
+                name: IgnoreIndexMetricWrapper(metric, ignore_index=self.hparams["ignore_index"])
+                for name, metric in metrics.items()
+            }
+
+        self.train_metrics = MetricCollection(wrap_metrics_with_ignore_index(instantiate_metrics()), prefix="train/")
+        self.val_metrics = MetricCollection(wrap_metrics_with_ignore_index(instantiate_metrics()), prefix="val/")
+        self.test_metrics = MetricCollection(wrap_metrics_with_ignore_index(instantiate_metrics()), prefix="test/")
+
+    def training_step(self, batch: Any, batch_idx: int, dataloader_idx: int = 0) -> Tensor:
+        """Compute the train loss and additional metrics.
+
+        Args:
+            batch: The output of your DataLoader.
+            batch_idx: Integer displaying index of this batch.
+            dataloader_idx: Index of the current dataloader.
+        """
+        x = batch["image"]
+        y = self.patchify(batch["mask"])
+
+        model_output: ModelOutput = self(x)
+        loss = self.train_loss_handler.compute_loss(model_output, y, self.criterion, self.aux_loss)
+        self.train_loss_handler.log_loss(self.log, loss_dict=loss, batch_size=x.shape[0])
+        y_hat = model_output.output
+        self.train_metrics(y_hat, y)
+        self.log_dict(self.train_metrics, on_epoch=True)
+
+        return loss["loss"]
+
+    def _do_plot_samples(self, batch_index):
+        if not self.plot_on_val:  # dont plot if self.plot_on_val is 0
+            return False
+
+        return (
+            batch_index < BATCH_IDX_FOR_VALIDATION_PLOTTING
+            and hasattr(self.trainer, "datamodule")
+            and self.logger
+            and not self.current_epoch % self.plot_on_val  # will be True every self.plot_on_val epochs
+            and hasattr(self.logger, "experiment")
+            and (hasattr(self.logger.experiment, "add_figure") or hasattr(self.logger.experiment, "log_figure"))
+        )
+
+    def validation_step(self, batch: Any, batch_idx: int, dataloader_idx: int = 0) -> None:
+        """Compute the validation loss and additional metrics.
+
+        Args:
+            batch: The output of your DataLoader.
+            batch_idx: Integer displaying index of this batch.
+            dataloader_idx: Index of the current dataloader.
+        """
+        x = batch["image"]
+        y = self.patchify(batch["mask"])
+       
+        model_output: ModelOutput = self(x)
+        loss = self.val_loss_handler.compute_loss(model_output, y, self.criterion, self.aux_loss)
+        self.val_loss_handler.log_loss(self.log, loss_dict=loss, batch_size=x.shape[0])
+
+        # Removing the time dimension
+        # TODO This should be less hard-coded
+        # However, it seems that all this part (for all the PixelWise Models) would not work
+        # for models with num_frames > 1.
+        y_hat = self.unpatchify(model_output.output)[:,:,0,...]
+        y = self.unpatchify(y)[:,:,0, ...]
```

Comment:
why is this logic now required? if its pre-training specific, It seems like it should be handled by the model. For example, if we are pre-training a CNN, this is not required